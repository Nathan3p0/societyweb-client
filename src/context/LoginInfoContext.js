import React from 'react';

const LoginInfoContext = React.createContext();

export default LoginInfoContext;
