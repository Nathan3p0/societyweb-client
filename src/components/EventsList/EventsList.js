import React, { Component } from 'react';
import EventListItem from '../EventsListItem/EventsListItem';
import LoginInfoContext from '../../context/LoginInfoContext';
import './EventsList.css'

class EventsList extends Component {
    static contextType = LoginInfoContext;

    state = {
        events: [],
        error: null
    }

    componentDidMount() {

    }

    render() {
        const { events } = this.context || [];
        let itemAmount = this.props.limit || events.length;
        const listItems = events.slice(0, itemAmount).map((event) =>
            <EventListItem key={event.id}
                id={event.id}
                date={event.event_date}
                name={event.event_name}
                time={event.event_time}
                location={event.event_location} />
        )
        return (
            <section className="events__list">
                <h3>Current Events:</h3>
                <ul className='responsiveTable'>
                    <EventsTableHeader />
                    {listItems}
                </ul>
            </section>
        );
    }
}

export default EventsList;


const EventsTableHeader = () => {
    return (
        <li className='tableHeader'>
            <div className='col col-1'>Event Date</div>
            <div className='col col-2'>Event Name</div>
            <div className='col col-3'>Event Time</div>
            <div className='col col-4'>Event Location</div>
        </li>
    );
}