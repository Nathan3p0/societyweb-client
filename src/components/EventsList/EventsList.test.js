import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import EventsList from './EventsList'
import LoginInfoContext from '../../context/LoginInfoContext';

describe('Events List Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(
            <LoginInfoContext.Provider value={{ events: [] }}>
                <EventsList />
            </LoginInfoContext.Provider>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(
            <LoginInfoContext.Provider value={{ events: [] }}>
                <EventsList />
            </LoginInfoContext.Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot()
    })
})