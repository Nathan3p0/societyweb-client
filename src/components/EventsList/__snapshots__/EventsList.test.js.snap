// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events List Component Rendering Tests renders to UI as expected 1`] = `
<section
  className="events__list"
>
  <h3>
    Current Events:
  </h3>
  <ul
    className="responsiveTable"
  >
    <li
      className="tableHeader"
    >
      <div
        className="col col-1"
      >
        Event Date
      </div>
      <div
        className="col col-2"
      >
        Event Name
      </div>
      <div
        className="col col-3"
      >
        Event Time
      </div>
      <div
        className="col col-4"
      >
        Event Location
      </div>
    </li>
  </ul>
</section>
`;
